### PR TITLE
187 Ten Tap

### LINKED ISSUE #187: Replace RichTextEditor with 10Tap Editor in EditNoteScreen.tsx
Description
We need to replace the existing RichTextEditor in the EditNoteScreen.tsx file with the 10Tap Editor.

Reason
The react-native-pell-rich-editor has not seen active updates for some time, and the community has noted that it might not be fully compatible with newer versions of React Native. While it's still usable, its development has slowed down, and there may be no official support for resolving bugs or keeping it up to date with future React Native releases.

Tasks
- [ ] Remove the current RichTextEditor component from the EditNoteScreen.tsx file.
- [ ] Install and integrate the 10Tap Editor into the project.
- [ ] Update necessary imports and dependencies to include the 10Tap Editor.
- [ ] Modify the component structure in EditNoteScreen.tsx to support text formatting using the 10Tap Editor.
- [ ] Test the new editor to ensure it supports the following features:
  - [ ] Text formatting (bold, italic, underline)
  - [ ] Bullet points and numbered lists
- [ ] Ensure compatibility with the existing state management system.
- [ ] Refactor any existing logic tied to the RichTextEditor to work with the 10Tap Editor.

Acceptance Criteria
- [ ] The 10Tap Editor is successfully implemented and replaces the RichEditor in the EditNoteScreen.tsx file.
- [ ] The editor supports basic text formatting options (bold, italic, underline) and list functionalities.
- [ ] All relevant tests should pass.
- [ ] The app should run on your local device/machine.

### PR DESCRIPTION
**Pull request by ademDurakovic**
Fixes #187 

Fixes: Replaced the richTextEditor to use the new tenTap library.

Why: TenTap is a more active library with help.

What got changed: Library import was added the editor was initialized and JSX components were added like the textpad/

### CODE REVIEW

File: lib/screens/EditNoteScreen.tsx [hunk 1]
```diff
@@ -30,6 +30,8 @@ import ToastMessage from 'react-native-toast-message';
 import { useTheme } from "../components/ThemeProvider";
 import LoadingModal from "../components/LoadingModal";
 import * as Location from 'expo-location';
+import { RichText, Toolbar, useEditorBridge } from "@10play/tentap-editor";
+
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx:34]: i see the addition for importing these tools

File: lib/screens/EditNoteScreen.tsx [hunk 2]
```diff
@@ -457,25 +443,22 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
...
+          </View>
+       
+        {/* Toolbar placed at the bottom */}
+        <View style={[NotePageStyles().toolBar]}>
+          <Toolbar
+            editor={editor}
+            style={NotePageStyles().container}
+            actions={['bold', 'italic', 'underline', 'bullet_list', 'blockquote', 'indent', 'outdent', 'close_keyboard' ]}
+          />
+        </View>
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx:461]: Nice addition to the code and watching out for anything that needed to be added.

**Review by teamomiamigo — COMMENTED** [review]
Overall, great addition to the file and in changing the text editor. Has made the code a little more simpler and cleaner.

**Comment by rcAsironman** [comment]
@ademDurakovic, you did a good job, but a few things should be improved in your work.

The Toolbar Alignment Issue: The toolbar is not aligned properly; it should be positioned right above the keyboard.

Your Output on Android:
Toolbar_current_output

Your Output on iOS:
Toolbar_current_output


Expected Output:
Mobile_editing


Theme Issue:
In dark mode, the editor should have a black background with white text. The opposite should apply for light mode.
Your Output on Android:

TextEditor_In_dark_mode


Your Output on iOS:

TextEditor_In_dark_mode

**Comment by ademDurakovic** [comment]
> @ademDurakovic, you did a good job, but a few things should be improved in your work.
> 
> The Toolbar Alignment Issue: The toolbar is not aligned properly; it should be positioned right above the keyboard.
> 
> Your Output on Android: Toolbar_current_output
> 
> Your Output on iOS: Toolbar_current_output
> 
> Expected Output: Mobile_editing
> 
> Theme Issue: In dark mode, the editor should have a black background with white text. The opposite should apply for light mode. Your Output on Android:
> 
> TextEditor_In_dark_mode
> Your Output on iOS:
> 
> TextEditor_In_dark_mode



> 

Here are how the new changes look:
Text is dynamically updated based on theme, and adjusted the toolbar.

image
image
image

**Review by rcAsironman — CHANGES_REQUESTED** [review]
@AdemDurakovic On Android, dark mode is not working as expected. The editor remains white while in dark mode. Please modify the code to set it to black.

File: lib/screens/EditNoteScreen.tsx [hunk 3]
```diff
@@ -69,7 +69,16 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
...
+  const [refreshEditor, setRefreshEditor] = useState(false);
+
+  // Pass the appropriate theme based on dark mode status
+  const editor = useEditorBridge({
+    initialContent: bodyText || "",
+    autofocus: true,
+  });
+  
+
+
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx:81]: good rework to make sure instantiating an editor

File: lib/screens/EditNoteScreen.tsx [hunk 4]
```diff
@@ -80,52 +89,112 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
...
+      ...Platform.select({  
+        android: {
+          flex: 1,
+          backgroundColor: theme.tertiaryColor, // Match the background color of your screen
+        },
+        ios: {
+          flex: 1,
+          backgroundColor: theme.tertiaryColor, // Match the background color of your screen
+        },
+      }),
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx:105]: good add, specifically for andriod to make sure that the colors are correct

File: lib/screens/EditNoteScreen.tsx [hunk 5]
```diff
@@ -80,52 +89,112 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
...
+        android: {
+          flex: 1,
+          minHeight: '100%', // Ensure it takes full height available
+          padding: 2,
+          borderWidth: 1,
+          borderColor: theme.primaryColor,
+          color: theme.primaryColor, // Set text color to white for dark mode
+        },
+
+      }),
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx:142]: proper for android coloring again, looks nice.

File: lib/screens/EditNoteScreen.tsx [hunk 6]
```diff
@@ -574,12 +574,14 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
         <View style={exampleStyles.richTextContainer}>
           <RichText 
             editor={editor} 
-            style={[exampleStyles.editor, {backgroundColor: Platform.OS == "android" && "white"}]} // Apply styling here
+            style={[exampleStyles.editor, {backgroundColor: Platform.OS == "android" && theme.tertiaryColor}]} // Apply styling here
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx]: nice, making sure it follows the theme

File: lib/screens/EditNoteScreen.tsx [hunk 7]
```diff
@@ -429,6 +429,21 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
...
+    const keyboardDidHideListener = Keyboard.addListener("keyboardDidHide", () => {
+      setIsKeyboardVisible(false);
+    });
+
+    return () => {
+      keyboardDidShowListener.remove();
+      keyboardDidHideListener.remove();
+    };
+  }, []);
+
```
Comment (teamomiamigo) [inline lib/screens/EditNoteScreen.tsx:446]: making sure the visibility is true, good job!

**Review by teamomiamigo — COMMENTED** [review]
Overall, great work on this, everything looks great and all the additions seem valid and correct! Great job!

File: lib/screens/EditNoteScreen.tsx [hunk 8]
```diff
@@ -80,52 +89,112 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
...
     },
   });
-  
-
-  // Inject and update the CSS in the editor whenever dark mode changes
+    
   useEffect(() => {
     if (editor) {
       editor.injectCSS(textColorCSS, 'text-color-style');
+      console.log("text color will be: " , theme.text);
```
Comment (rcAsironman) [inline lib/screens/EditNoteScreen.tsx:194]: please remove the log statements.

File: lib/screens/EditNoteScreen.tsx [hunk 9]
```diff
@@ -356,6 +425,11 @@ const EditNoteScreen: React.FC<EditNoteScreenProps> = ({
     }
   };
 
+  if (isDarkmode && editor) {
+    editor.injectCSS(textColorCSS, 'text-color-style');
+    console.log("text color will be: #F7F8F9 from IF state");
```
Comment (rcAsironman) [inline lib/screens/EditNoteScreen.tsx:430]: Remove any log statements before committing.

**Review by rcAsironman — CHANGES_REQUESTED** [review]
@ademDurakovic I like the changes you've made. But, I came across an issue: the toolbar is not aligned properly.

Issue:

On iOS 12, the toolbar is not showing. It should work on all devices.
IMG_0223

**Review by yashb196 — CHANGES_REQUESTED** [review]
Overall Good Effort however when you enter EditNoteScreen you see this error on the console log 

**### Warning: React has detected a change in the order of Hooks called by EditNoteScreen. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useState                   useState
2. useState                   useState
3. useState                   useState
4. useState                   useState
5. useState                   useState
6. useState                   useState
7. useState                   useState
8. useState                   useState
9. useState                   useState
10. useRef                    useRef
11. useState                  useState
12. useState                  useState
13. useState                  useState
14. useState                  useState
15. useState                  useState
16. useState                  useState
17. useState                  useState
18. useState                  useState
19. useRef                    useRef
20. useState                  useState
21. useState                  useState
22. useState                  useState
23. useContext                useContext
24. useState                  useState
25. useRef                    useRef
26. useRef                    useRef
27. useRef                    useRef
28. useRef                    useRef
29. useMemo                   useMemo
30. useMemo                   useMemo
31. useEffect                 useEffect
32. useEffect                 useEffect
33. useState                  useState
34. useState                  useState
35. useEffect                 useEffect
36. useEffect                 useEffect
37. useEffect                 useEffect
38. useState                  useState
39. useState                  useState
40. useRef                    useRef
41. useState                  useState
42. useEffect                 useEffect
43. useContext                useContext
44. useContext                useContext
45. useContext                useContext
46. useContext                useContext
47. useContext                useContext
48. useContext                useContext
49. useContext                useContext
50. useContext                useContext
51. useContext                useContext
52. useContext                useContext
53. useContext                useContext
54. useContext                useContext
55. undefined                 useContext
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    in EditNoteScreen (created by EditNote)
    in EditNote (created by SceneView)
    in StaticContainer
    in EnsureSingleNavigator (created by SceneView)
    in SceneView (created by CardContainer)
    in RCTView (created by View)
    in View (created by CardContainer)
    in RCTView (created by View)
    in View (created by CardContainer)
    in RCTView (created by View)
    in View
    in CardSheet (created by Card)
    in RCTView (created by View)
    in View (created by Animated(View))
    in Animated(View) (created by PanGestureHandler)
    in PanGestureHandler (created by PanGestureHandler)
    in PanGestureHandler (created by Card)
    in RCTView (created by View)
    in View (created by Animated(View))
    in Animated(View) (created by Card)
    in RCTView (created by View)
    in View (created by Card)
    in Card (created by CardContainer)
    in CardContainer (created by CardStack)
    in RCTView (created by View)
    in View (created by MaybeScreen)
    in MaybeScreen (created by CardStack)
    in RCTView (created by View)
    in View (created by MaybeScreenContainer)
    in MaybeScreenContainer (created by CardStack)
    in RCTView (created by View)
    in View (created by Background)
    in Background (created by CardStack)
    in CardStack (created by HeaderShownContext)
    in RCTView (created by View)
    in View (created by SafeAreaProviderCompat)
    in SafeAreaProviderCompat (created by StackView)
    in RCTView (created by View)
    in View (created by GestureHandlerRootView)
    in GestureHandlerRootView (created by StackView)
    in StackView (created by StackNavigator)
    in PreventRemoveProvider (created by NavigationContent)
    in NavigationContent
    in Unknown (created by StackNavigator)
    in StackNavigator (created by HomeStack)
    in HomeStack (created by SceneView)
    in StaticContainer
    in EnsureSingleNavigator (created by SceneView)
    in SceneView (created by BottomTabView)
    in RCTView (created by View)
    in View (created by Screen)
    in RCTView (created by View)
    in View (created by Background)
    in Background (created by Screen)
    in Screen (created by BottomTabView)
    in RCTView (created by View)
    in View (created by ResourceSavingScene)
    in RCTView (created by View)
    in View (created by ResourceSavingScene)
    in ResourceSavingScene (created by MaybeScreen)
    in MaybeScreen (created by BottomTabView)
    in RCTView (created by View)
    in View (created by MaybeScreenContainer)
    in MaybeScreenContainer (created by BottomTabView)
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider (created by SafeAreaProviderCompat)
    in SafeAreaProviderCompat (created by BottomTabView)
    in BottomTabView (created by BottomTabNavigator)
    in PreventRemoveProvider (created by NavigationContent)
    in NavigationContent
    in Unknown (created by BottomTabNavigator)
    in BottomTabNavigator (created by AppNavigator)
    in EnsureSingleNavigator
    in BaseNavigationContainer
    in ThemeProvider
    in NavigationContainerInner (created by AppNavigator)
    in AppNavigator (created by App)
    in ThemeProvider (created by App)
    in Provider (created by App)
    in App (created by withDevTools(App))
    in withDevTools(App)
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in main(RootComponent)**
    
    this will break the application I would ask you to have a look at this and fix it before merging

**Review by rcAsironman — CHANGES_REQUESTED** [review]
@ademDurakovic On Android, two toolbars are being rendered. Please check iOS as well.

</img>

**Review by rcAsironman — APPROVED** [review]

**Review by yashb196 — APPROVED** [review]

**Pull request merged**